### PR TITLE
Vscode ext should update default connection

### PR DESCRIPTION
**Pull request by sdedic**
The vscode "Database" view queried NBLS for the default DB just on startup, then it retained the connection node's ID, and only allowed vscode > NBLS change propagation. We need - for example when defining a new DB - that the NBLS's default connection is reflected back in the vscode. That will help if no connection is present and a new (first) one is defined that becomes the default. May also help in cases when other command(s) redefine the default connection internally.

I've changed DB module since there was a bug in that it offered the 1st connection as a default if none was specifically set - but was willing to change the default 'on the fly' if a new connection appeared, so the default could shift unexpectedly. Case when there's a preferred connection name stored, but the definition is somehow missing was also mishandled.

### CODE REVIEW

File: ide/db/src/org/netbeans/modules/db/explorer/ConnectionList.java [hunk 1]
```diff
@@ -273,32 +278,57 @@ public void setPreferredConnection(DatabaseConnection c) {
...
+            }
+        }
+        if (prefChanged) {
+            fireListeners();
+        }
+        return p;
+    }
+    
+    private static DatabaseConnection findConnection(String name, DatabaseConnection[] conns) {
         for (int i = 0; i < conns.length; i++) {
```
Comment (lbownik) [inline ide/db/src/org/netbeans/modules/db/explorer/ConnectionList.java:327]: for-each loop maybe?

Comment (jhorvath) [reply ide/db/src/org/netbeans/modules/db/explorer/ConnectionList.java:327]: I don't see the problem, it's a matter of personal preference.

**Review by jhorvath — APPROVED** [review]
Looks good

File: ide/db/src/org/netbeans/modules/db/explorer/ConnectionList.java (hunk 1 above)
Comment (sdedic) [reply ide/db/src/org/netbeans/modules/db/explorer/ConnectionList.java:327]: I usually use for-loops in these situations ... this part was actually extracted from the older code that is obviously no excuse for not updating the style. I usually incorporate 'style issues' whenever some func/perf ("real") issue is found and a new commit has to be made.

**Pull request merged**